src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/RObjectEntry.java

```diff
@@ -96,9 +98,29 @@ public String getDisplayValue()
                       rObject.getDescription().trim() :
                       val;
    }
+   
+   private boolean rObjectHasDataClass()
+   {
+      JsArrayString classes = rObject.getClazz();
+      for (int i = 0, n = classes.length(); i < n; i++)
+         if (DATA_CLASSES.contains(classes.get(i)))
+            return true;
+      return false;
+   }
 
    public static final String NO_VALUE = "NO_VALUE";
-
+   
+   private static final Set<String> DATA_CLASSES = new HashSet<String>();
+   static {
+      DATA_CLASSES.addAll(ListUtil.create(
```

Comment:
Good catch; I think you're right. (I initially thought that `class()` would return the base class for e.g. S4 `DataFrame` objects, thereby ensuring that the plain old `data.frame` check would succeed, but that is not the case -- only `DataFrame` is returned.)